include/SDL3/SDL_intrin.h

```diff
@@ -93,26 +93,42 @@ _m_prefetch(void *__P)
 #endif
 #endif /* compiler version */
 
+#if defined(__clang__) && defined(__has_attribute)
+# if __has_attribute(target)
+# define SDL_HAS_TARGET_ATTRIBS
+# endif
+#elif defined(__GNUC__) && (__GNUC__ + (__GNUC_MINOR__ >= 9) > 4) /* gcc >= 4.9 */
+# define SDL_HAS_TARGET_ATTRIBS
+#endif
+
+#ifdef SDL_HAS_TARGET_ATTRIBS
+# define SDL_TARGETING(x) __attribute__((target(x)))
+#else
+# define SDL_TARGETING(x)
+#endif
+
 #if defined(__loongarch_sx) && !defined(SDL_DISABLE_LSX)
 #include <lsxintrin.h>
 #endif
 #if defined(__loongarch_asx) && !defined(SDL_DISABLE_LASX)
 #include <lasxintrin.h>
 #endif
-#if defined(__AVX__) && !defined(SDL_DISABLE_AVX)
+#if defined(__x86_64__) || defined(_M_X64) || defined(__i386__) || defined(_M_IX86)
+#if (defined(__AVX__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_AVX)
 #include <immintrin.h>
 #endif
-#if defined(__MMX__) && !defined(SDL_DISABLE_MMX)
+#if (defined(__MMX__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_MMX)
 #include <mmintrin.h>
 #endif
```

Comment:
These checks are duplicated in `SDL_internal.h`. Perhaps we can introduce a macro informing people the header is available?
e.g.
```suggestion
#if (defined(__MMX__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_MMX)
#define SDL_MMX_INTRINSICS
#include <mmintrin.h>
#endif
```

People can then easily do the following in their code:
```c
#include <SDL3/SDL_intrin.h>

void float_add(float *__restrict__ dest, float *__restrict__ a, float *__restrict__ b, size_t count) {
#if defined(SDL_AVX512_INTRINSICS)
  for (; count > 16; count -= 16, dest += 16, a += 16, b += 16) {
    __m512 ra = _mm512_loadu_ps(a);
    __m512 rb = _mm512_loadu_ps(b);
    __m512 rd = _mm512_add_ps(ra, rb);
    _mm512_storeu_ps(dest, rd);
  }
#elif defined(SDL_AVX_INTRINSICS)
  for (; count > 8; count -= 8, dest += 8, a += 8, b += 8) {
    __m256 ra = _mm256_loadu_ps(a);
    __m256 rb = _mm256_loadu_ps(b);
    __m256 rd = _mm256_add_ps(ra, rb);
    _mm256_storeu_ps(dest, rd);
  }
#elif defined(SDL_SSE_INTRINSICS)
  for (; count > 4; count -= 4, dest += 4, a += 4, b += 4) {
    __m128 ra = _mm_loadu_ps(a);
    __m128 rb = _mm_loadu_ps(b);
    __m128 rd = _mm_add_ps(ra, rb);
    _mm_storeu_ps(dest, rd);
  }
#endif
  for (; count; --count, ++dest, ++a, ++b) {
    *dest = *a + *b;
  }
}
```

Reply:
Feel free to commit to my branch

Reply:
Will do.
With these changes, we can perhaps remove the `#define __SSE__`'s and similar in `SDL_intrin.h`.
I feel we shouldn't define double underscore macros/variables: that's the terrain of compiler/stdlib implementers.

Reply:
I kept it, but is this workaround still needed?
https://github.com/libsdl-org/SDL/blob/08d5c1386e8eb468f0c66b5511c2d334e6ae0b22/include/SDL3/SDL_intrin.h#L33-L44

Do we still support VS2005? In that year, gcc 3.4.3 and gcc 4.0.0 were released.

Reply:
> I kept it, but is this workaround still needed?

Really don't know. clang-cl testing would be needed 

> Do we still support VS2005? In that year, gcc 3.4.3 and gcc 4.0.0 were released.

For building SDL itself, no. However it's a public header, so we don't know what the user will be doing.

Reply:
No, we don't still support VS2005. VS2010 is the earliest we support.